Css color consistency

Arnei opened the pull request:
Includes #1400. #1400 Should be reviewed and merged first.

This PR replaces the "hardcoded" color values that are flying around everywhere in our scss files with the color variables defined in `_variables.scss`. Color values that are not represented by our variables are either added to the variables file, or replaced with a variable with a similar value.

Goals:
- Make colors easier to work with. Having our colors all grouped in one place should make it easier to e.g. implement a dark mode.
- Reduce the number of different colors. We really should not need 50 different shades of grey (number approximated).
- Increase contrast. I used the opportunity to slightly change some colors to make the ui more readable.

### How to test this

This touches every element that has color, so unfortunately pretty much everything. Quickly click through everything ui element you can find and check if any colors appear "off".

Arnei committed:
Align dropdown colors

Make more dropdowns have the same background and text color
for hovered options.

Arnei committed:
Adjust header and button colors

Arnei committed:
Adjust published cell

Make the published cell in the events table look like other cells with
similar appearance.

Arnei committed:
Progress bar leftover colors

Arnei committed:
Remove unused css from wizard stepper

Arnei committed:
Turn "OK" button colors to variables

Arnei committed:
Normalize table border colors

Arnei committed:
Normalize various table colors

Arnei committed:
Normalize statistics dropdown colors

Arnei committed:
Normalize colors in various modals

Arnei committed:
Fix chevron color

In the table heeader, fix both sorting chevrons being colored
as "active", instead of only one.
Also make the active chevron darker to be more easily
recognizable.

Arnei committed:
Merge branch 'css-more-cleanup' into css-color-consistency

Arnei committed:
Merge remote-tracking branch 'upstream/develop' into css-color-consistency

ferishili commented on src/styles/mixins/_button.scss:
I noticed there's a whole block of code commented out. Is that intentional?
If so, could you please add a short comment above it explaining why it's being kept? Also, it would be better to use a proper block comment instead of multiple single-line comments for readability.

Arnei replied:
Thanks for the review! I added a comment on why I thought this might be worth keeping, even if commented out. Will also take a look at the calendar.

ferishili reviewed (changes requested):
Thanks @Arnei,
Your changes look fine, however a few things need your attention!

First and most important one is:
While running the UI click-through tests, I noticed the contrast in the calendar (month and year selection) looks off, as shown below. Could you please take a look at that as well?

<img width="802" height="725" alt="Image" src="https://github.com/user-attachments/assets/e0c7d421-1483-43a2-834c-2dac9ad86ab7" />

Arnei committed:
Add explanation to commented out code

Arnei committed:
Make text darker for month/year in datepicker

White on light background is hard to read. Black is more in line with
the other text and provides better contrast.

Arnei committed:
Improve month/year picker background contrast

Upon further feedback, this changes the text color
back to white, but improves the contrast of the background instead.

ferishili reviewed (approved):
Everything look fine, thanks @Arnei
